server/index.js

```diff
@@ -12,4 +12,9 @@ app.get('/api/productPreview/:productId/:colourName', (req, res) => {
     .catch(err => res.json(err));
 });
 
+app.get('/api/productPreview/:productId', (req, res) => {
+  productData(req.params.productId)
+    .then(results => res.json(results));
+});
+
```

Comment:
Couldn't find anything that needed changes so I just re-wrote your get request using async/await 🤣 

app.get('/api/productPreview/:productId', async (req, res) => {
  const results = await productData(req.params.productId)
  res.json(results));
});